src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecuteConfigTest.java

```diff
@@ -33,38 +33,57 @@
-    BigQueryExecute.Config config = new BigQueryExecute.Config("standard",
-                                                               "select * from dataset.table where id=1",
-                                                               "batch");
-
-    FieldSetter.setField(config, GCPConfig.class.getDeclaredField("project"), "test_project");
+    BigQueryExecute.Config config = getConfig();
 
     MockFailureCollector failureCollector = new MockFailureCollector();
     BigQuery bigQuery = mock(BigQuery.class);
-    when(bigQuery.create(ArgumentMatchers.any(JobInfo.class))).thenReturn(ArgumentMatchers.any());
+    when(bigQuery.create(ArgumentMatchers.any(JobInfo.class))).thenReturn(ArgumentMatchers.any(Job.class));
 
     config.validateSQLSyntax(failureCollector, bigQuery);
     Assert.assertEquals(0, failureCollector.getValidationFailures().size());
   }
 
+  private BigQueryExecute.Config getConfig() throws NoSuchFieldException {
+    BigQueryExecute.Config.Builder builder = BigQueryExecute.Config.builder();
+    return builder
+            .setDialect("standard")
+            .setSql("select * from dataset.table where id=1")
+            .setMode("batch")
+            .setProject("test_project")
+            .build();
+  }
+
   @Test
-  public void testBigQueryExecuteInvalidSQL() throws Exception {
-    String errorMessage = "Invalid sql";
-    BigQueryExecute.Config config = new BigQueryExecute.Config("standard",
-                                                               "selcet * fro$m dataset-table where id=1",
-                                                               "batch");
+  public void testBigQueryExecuteSQLWithNonExistentResource() throws Exception {
+    String errorMessage = "Resource was not found. Please verify the resource name. If the resource will be created " +
+      "at runtime, then update to use a macro for the resource name. Error Message returned: ";
+    BigQueryExecute.Config config = getConfig();
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    BigQuery bigQuery = mock(BigQuery.class);
+    when(bigQuery.create(ArgumentMatchers.any(JobInfo.class))).thenThrow(new BigQueryException(404, ""));
 
-    FieldSetter.setField(config, GCPConfig.class.getDeclaredField("project"), "test_project");
+    config.validateSQLSyntax(failureCollector, bigQuery);
+    Log.warn("size : {}", failureCollector.getValidationFailures().size());
+    Assert.assertEquals(1, failureCollector.getValidationFailures().size());
+    Assert.assertEquals(String.format("%s.", errorMessage),
+            failureCollector.getValidationFailures().get(0).getMessage());
+    Assert.assertEquals("sql",
+            failureCollector.getValidationFailures().get(0).getCauses().get(0).getAttribute("stageConfig"));
+  }
 
+  @Test
+  public void testBigQueryExecuteInvalidSQL() throws Exception {
+    String errorMessage = "Invalid sql";
+    BigQueryExecute.Config config = getConfig();
```

Comment:
`getConfig()` will return a config with valid SQL query or am I missing something? Though we are throwing an error later but it looks confusing! Can we pass the `sql` param to the `getConfig(String sql)` to make it look better?

Reply:
yes I refactored this in the next rev. I initially mixed up the sql query at #L50 to be used for both valid and invalid queries.